api/v1alpha1/webhook_helpers.go

```diff
@@ -0,0 +1,280 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+
+	"github.com/nutanix-cloud-native/ndb-operator/api"
+	"github.com/nutanix-cloud-native/ndb-operator/common"
+	"github.com/nutanix-cloud-native/ndb-operator/common/util"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// Get specific implementation of the DBProvisionRequestAppender interface based on the provided databaseType
+func getDatabaseWebhookHandler(database *Database) DatabaseWebhookHandler {
+	if database.Spec.IsClone {
+		return &CloningHandler{}
+	} else {
+		return &ProvisoningHandler{}
+	}
+}
+
+// +kubebuilder:object:generate:=false
+type DatabaseWebhookHandler interface {
+	// Default logic
+	defaulter(databaseSpec *DatabaseSpec)
+	// Validates creation (after defaulting)
+	validateCreate(databaseSpec *DatabaseSpec, errors *field.ErrorList, instancePath *field.Path)
+}
+
+// +kubebuilder:object:generate:=false
+// Implements webhook.Validator, webhook.Defaulter
+type CloningHandler struct{}
+
+// +kubebuilder:object:generate:=false
+// Implements webhook.Validator, webhook.Defaulter
+type ProvisoningHandler struct{}
+
+func (v *CloningHandler) defaulter(spec *DatabaseSpec) {
```

Comment:
Possibly better name would be CloningWebhookHandler & ProvisioningWebhookHandler

Reply:
Sounds good.